oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+   former are easier to reason about because they don't support arbitrary
+   code evaluation. See `Reasons for rejecting Mako`_ for a more detailed
+   explanation.
+
+   **Exception**: Mako templates should be used within edx-platform for
+   consistency with the existing code base. Unfortunately there has been too
+   much investment in Mako to consider replacing it at this point.
+
+2. **ECMAScript 2015 should be the supported version of JavaScript**
+
+   Rationale: ECMAScript is the standardized version of JavaScript,
+   and the most recently ratified version is ECMAScript 2015. It is
+   commonly referred to by ES2015, or by ES6 (its original name). Most
+   browsers now provide native support for ES2015, but it is generally
+   recommended to use `Babel`_ to transpile ES2015 code into ES5 which is
+   universally supported.
+
+3. **JavaScript libraries should be installed via npm**
+
+   Rationale: It is important that JavaScript libraries are kept
+   up-to-date, and one key component is to make it as simple as possible
+   to perform upgrades. Projects that use npm maintain all of their
+   library requirements in a single package.json file, which can be
+   easily changed as the versions change.
+
+4. **JavaScript code should follow the edX ESLint configuration**
+
+   Rationale: In order to standardize and enforce Open edX's JavaScript
+   coding style across multiple codebases, edX has published an ESLint
+   configuration that provides an enforceable specification. The
+   `edX ESLint Config`_ is made available as an npm package that can be
+   installed into any Open edX package.
+
+5. **Front-end interactivity should be added using Backbone**
+
+   Rationale: There are a large number of JavaScript libraries for
+   building rich client applications, but many of them are too
+   rigid for a flexible, extensible codebase like Open edX. Backbone
+   has the major benefit that it can be added incrementally to existing
+   pages, no matter the framework that is used. It also provides a clean
+   programming model that enforces a clean separation between models
+   and views.
+
+   Note: Backbone is a somewhat old technology in the JavaScript world
+   and has seen a rapid drop off in usage within the community. See
+   :ref:`JavaScript Frameworks` below for the open question of which
+   of the many alternate frameworks should become the new Open edX standard.
+   For now, Backbone is the library of choice because so much of the
+   Open edX codebase uses it.
+
+6. **JavaScript dependencies should be declared via RequireJS**
+
+   Rationale: Open edX is a large, complicated code base, and it is
+   getting more tangled over time. RequireJS allows code to declare
+   the dependencies it requires, rather than forcing every page to load
+   all possible libraries. It also supports intelligent bundling using
+   RequireJS Optimizer, which can package together all required libraries
+   into a single file for faster performance.
```

Comment:
I want to see the ECOM Team move toward Webpack. Our IDAs with static assets waste a lot of time moving unnecessary files around due to the nature of Django-compressor and RequireJS.